### PR TITLE
Accessibility updates [uswds-site 1227]

### PR DESCRIPTION
**Pull request by thisisdano**
## Summary
This adjusts some of the component code templates to allow the final output code to pass accessibility tests on our website

## Requirements
This is a companion to https://github.com/uswds/uswds-site/pull/1227
Approve and merge this PR first. Then link the `uswds-site` PR to the develop branch. 

## Details
See code comments

### CODE REVIEW

File: src/components/button-groups/button-groups.njk [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <ul class="usa-button-group">
   <li class="usa-button-group__item">
-    <a href="#" class="usa-button usa-button--outline">Back</a>
+    <a href="javascript:void(0);" class="usa-button usa-button--outline">Back</a>
```
Comment (thisisdano) [inline src/components/button-groups/button-groups.njk:3]: Avoid em empty hash links, use `javascript:void(0);` instead

File: src/components/banner/banner.config.yml [hunk 2]
```diff
@@ -36,6 +38,8 @@ variants:
           Un <strong>candado</strong> ($LOCK_IMG) o <strong>https://</strong> significa que usted se conectó de forma segura a un sitio web $TLD.  Comparta información sensible sólo en sitios web oficiales y seguros.
   - name: dot-mil
     context:
+      banner:
+        aria_label: "Official government website,"
```
Comment (thisisdano) [inline src/components/banner/banner.config.yml:42]: ... and use trailing commas as a hack to provide legible ARIA labels but still make them unique

File: src/components/summary-box/summary-box.njk [hunk 3]
```diff
@@ -1,21 +1,21 @@
-<div class="usa-summary-box" role="complementary">
+<div class="usa-summary-box">
```
Comment (thisisdano) [inline src/components/summary-box/summary-box.njk:1]: Remove `complementary` role for content inside a `main` element

File: src/components/summary-box/summary-box.njk [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
   <div class="usa-summary-box__body">
     <h3 class="usa-summary-box__heading">Key information</h3>
     <div class="usa-summary-box__text">
       <ul class="usa-list">
         <li>If you are under a winter storm warning,
-          <a class="usa-summary-box__link" href="#usa-anchor-find-shelter">find shelter</a>
+          <a class="usa-summary-box__link" href="javascript:void(0);">find shelter</a>
```
Comment (thisisdano) [inline src/components/summary-box/summary-box.njk:7]: Don't use fake anchor links

File: src/components/table/table--scrollable.njk [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<div class="usa-table-container--scrollable">
+<div class="usa-table-container--scrollable" tabindex="0">
```
Comment (thisisdano) [inline src/components/table/table--scrollable.njk:1]: Assure that scrollable elements are focusable

File: src/components/type/lists.njk [hunk 6]
```diff
@@ -1,6 +1,6 @@
   <div class="grid-row grid-gap">
     <div class="mobile-lg:grid-col-4">
-      <h6 class="margin-0">Unordered list</h6>
+      <h3 class="site-preview-heading margin-0">Unordered list</h3>
```
Comment (thisisdano) [inline src/components/type/lists.njk:3]: Use proper heading level

File: src/components/type/typesetting.njk [hunk 7]
```diff
@@ -1,99 +1,97 @@
-<div class="usa-prose">
-  <div class="usa-line-length-example usa-prose">
-    <h6 class="usa-heading-alt">Line length</h6>
-    <section class="typography-example usa-prose margin-top-1">
-      <p><strong>75 characters (68ex) max-width:</strong> Yosemite National Park is set within California’s Sierra Nevada mountains. It’s famed for its giant, ancient sequoias, and for Tunnel View, the iconic vista of towering Bridalveil Fall and the granite cliffs of El Capitan and Half Dome.</p>
-    </section>
-  </div>
+<div class="usa-line-length-example">
+  <h3 class="site-preview-heading margin-top-0">Line length</h3>
```
Comment (thisisdano) [inline src/components/type/typesetting.njk:2]: Use proper heading level and only include sample text within the `usa-prose`

**Review by mejiaj — COMMENTED** [review]
A few questions on Identifier and Breadcrumb href. Everything else looks good.

File: src/components/button-groups/button-groups.njk (hunk 1 above)
Comment (thisisdano) [reply src/components/button-groups/button-groups.njk:3]: I guess so... I wonder why those weren't flagged as errors....

File: src/components/summary-box/summary-box.njk (hunk 3 above)
Comment (thisisdano) [reply src/components/summary-box/summary-box.njk:1]: It does not have an explicit `region` now. I can give it one

File: src/components/identifier/identifier.config.yml [hunk 8]
```diff
@@ -6,8 +6,8 @@ context:
   masthead:
     label:
       content:
-        en: Agency identifier
-        es: Identificador de la agencia
+        en: Agency identifier,
+        es: Identificador de la agencia,
     description:
       label:
         content:
```
[the remark this answers is not in the document]
Comment (thisisdano) [reply src/components/identifier/identifier.config.yml:13]: Yes

File: src/components/navigation/sidenav.config.yml [hunk 9]
```diff
@@ -28,6 +28,7 @@ variants:
   - name: one-level
     context:
       sidenav:
+        aria: Secondary navigation,
```
[the remark this answers is not in the document]
Comment (thisisdano) [reply src/components/navigation/sidenav.config.yml:31]: Yup

**Review by mejiaj — APPROVED** [review]

**Pull request merged**